### PR TITLE
ElastiCache Redis

### PR DESCRIPTION
**Pull request by Stretch96**
* Conditionally launch ElastiCache Redis
* Allows chosing either Cluster or Serverless
* ElastiCache also supports Memcache, but this module currently only supports Redis

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -458,6 +458,51 @@ variable "infrastructure_rds" {
...
+    cluster_node_count       = optional(number, null)
+    serverless_max_storage   = optional(number, null)
+    serverless_max_ecpu      = optional(number, null)
+    snapshot_retention_limit = optional(number, null)
+  })
+}
+
+variable "infrastructure_elasticache" {
+  description = <<EOT
+    Map of RDSs (The key will be the rds name). Values in here will override `infrastructure_rds_defaults` values if set."
```
Comment (rjw1) [inline variables.tf:478]: its not RDS.

**Review by rjw1 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (Stretch96) [reply variables.tf]: Ah ... Updated 👍 ...

File: variables.tf [hunk 2]
```diff
@@ -458,6 +458,51 @@ variable "infrastructure_rds" {
...
+    serverless_max_ecpu      = optional(number, null)
+    snapshot_retention_limit = optional(number, null)
+  })
+}
+
+variable "infrastructure_elasticache" {
+  description = <<EOT
+    Map of Elasticaches (The key will be the elasticache name). Values in here will override `infrastructure_elasticache_defaults` values if set."
+    {
+      rds-name = {
```
Comment (rjw1) [inline variables.tf:480]: this line to :)

**Review by rjw1 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (Stretch96) [reply variables.tf]: Updated 👍

**Review by rjw1 — APPROVED** [review]
looks fine

**Pull request merged**